File: JPLearningHub_Client/lib/models/user_model.dart
import 'package:json_annotation/json_annotation.dart';
part 'user_model.g.dart';

@JsonSerializable()
class UserModel {
  String firstName;
  String lastName;
  String avatarKey;
  String avatarUrl;
  String userId;
  String timeZone;
  String userType;

  UserModel(
      {required this.firstName,
      required this.lastName,
      required this.avatarKey,
      required this.avatarUrl,
      required this.userId,
      required this.timeZone,
      required this.userType});

  factory UserModel.fromJson(Map<String, dynamic> json) =>
      _$UserModelFromJson(json);
  Map<String, dynamic> toJson() => _$UserModelToJson(this);

  bool isTeacher() {
    return userType == "Teacher";
  }
}

File: JPLearningHub_Server/src/JPLearningHub_Server/Models/UserModel.cs
using JPLearningHub_Server.Entities;

namespace JPLearningHub_Server.Models
{
    public class UserModel
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string AvatarKey { get; set; } = string.Empty;
        public string AvatarUrl { get; set; } = string.Empty;
        public string UserId { get; set; } = string.Empty;
        public string TimeZone { get; set; } = string.Empty;
        public string UserType { get; set; } = string.Empty;

        public static UserModel FromUser(User user, string avatarUrl)
        {
            return new UserModel
            {
                FirstName = user.FirstName,
                LastName = user.LastName,
                AvatarKey = user.AvatarKey,
                AvatarUrl = avatarUrl,
                UserId = user.UserId,
                TimeZone = user.TimeZone,
                UserType = user.UserType
            };
        }
    }
}
